core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/worker/statistics.proto

```diff
@@ -21,10 +21,14 @@ enum WorkerState {
 }
 
 message WorkerStatistics {
-    WorkerState worker_state = 1 [(scalapb.field).no_box = true];
-    int64 input_tuple_count = 2;
-    int64 output_tuple_count = 3;
-    int64 data_processing_time = 4;
-    int64 control_processing_time = 5;
-    int64 idle_time = 6;
+    map<int32, int64> input_tuple_count = 1;
+    map<int32, int64> output_tuple_count = 2;
```

Comment:
I assume the key of the map is portId. Can we use `PortIdentity` instead of `int32` for the keys?

Reply:
There is a limitation of Protocol Buffer (protobuf) regarding the types that can be used as keys in a map. In protobuf, the types `float`, `double`, `bytes`, `message`, and `enum` are not allowed as map keys. Since, `PortIdentity` is a `message` type, we cannot use it for the keys.

Reply:
yes this is known limitation of protobuf map which is an extension. you can use `repeated` keyword (the old way) in this case https://stackoverflow.com/questions/39362340/key-in-map-fields-cannot-be-float-double-bytes-or-message-types

Reply:
Fixed.